src/components/SubmissionProgress.tsx

```diff
@@ -26,7 +26,26 @@ function Heading(props: { children: React.ReactNode }) {
   )
 }
 
-function SubmissionProgress(props: { onClose?: () => void; promise: Promise<any> }) {
+export type SubmissionType = "default" | "multi-sig" | "stellarguard"
```

Comment:
[Nitpick, could be merged anyway]
Might lead to slightly better code if we just define an object that maps from `SubmissionType` to `string` and make `SubmissionType` an enum:

```ts
enum SubmissionType {
  default,
  multisig,
  stellarguard
}

const successMessages: { [type: SubmissionType]: string } = {
  [SubmissionType.default]: "Successful",
  [SubmissionType.multisig]: "Waiting for missing signatures",
  [SubmissionType.stellarguard]: "Waiting for StellarGuard authorization"
}
```

And then further down
```diff
- <Heading>{getSuccessMessage(props.type)}</Heading>
+ <Heading>{successMessages[props.type]}</Heading>
```